src/app/pages/dashboard/services/widget-resources.service.spec.ts

```diff
@@ -20,24 +30,202 @@ const apps = [
   let spectator: SpectatorService<WidgetResourcesService>;
+  let testScheduler: TestScheduler;
+
+  const nics: NetworkInterface[] = [
+    {
+      id: 'ens1',
+      name: 'ens1',
+      fake: false,
+      type: NetworkInterfaceType.Physical,
+      state: {
+        name: 'ens1',
+        orig_name: 'ens1',
+        description: 'ens1',
+        mtu: 1500,
+        cloned: false,
+        flags: [
+          NetworkInterfaceFlag.Broadcast,
+          NetworkInterfaceFlag.Multicast,
+          NetworkInterfaceFlag.Up,
+          NetworkInterfaceFlag.Running,
+          NetworkInterfaceFlag.LowerUp,
+        ],
+        nd6_flags: [
+          'HOMEADDRESS',
+        ],
+        capabilities: [
+          'tx-scatter-gather',
+          'tx-checksum-ipv4',
+          'tx-vlan-hw-insert',
+          'rx-vlan-hw-parse',
+          'tx-generic-segmentation',
+          'rx-gro',
+          'tx-tcp-segmentation',
+          'rx-checksum',
+        ],
+        link_state: LinkState.Up,
+        media_type: 'Ethernet',
+        media_subtype: 'autoselect',
+        active_media_type: 'Ethernet',
+        active_media_subtype: '100Mb/s MII',
+        supported_media: [
+          '10baseT/Half',
+          '10baseT/Full',
+          '100baseT/Half',
+          '100baseT/Full',
+        ],
+        media_options: null,
+        link_address: '52:54:00:2c:4d:f9',
+        aliases: [
+          {
+            type: NetworkInterfaceAliasType.Inet,
+            address: '10.220.38.162',
+            netmask: 22,
+            broadcast: '10.220.39.255',
+          },
+          {
+            type: NetworkInterfaceAliasType.Inet,
+            address: '10.220.39.70',
+            netmask: 32,
+            broadcast: '10.220.39.70',
+          },
+          {
+            type: NetworkInterfaceAliasType.Inet6,
+            address: 'fe80::5054:ff:fe2c:4df9',
+            netmask: 64,
+            broadcast: 'fe80::ffff:ffff:ffff:ffff',
+          },
+          {
+            type: NetworkInterfaceAliasType.Link,
+            address: '52:54:00:2c:4d:f9',
+          },
+        ],
+        vrrp_config: [
+          {
+            address: '10.220.39.70',
+            state: 'MASTER',
+          },
+        ],
+      },
+      aliases: [
+        {
+          type: NetworkInterfaceAliasType.Inet,
+          address: '10.220.38.162',
+          netmask: 22,
+        },
+      ],
+      ipv4_dhcp: false,
+      ipv6_auto: false,
+      description: '',
+      mtu: null,
+      failover_critical: true,
+      failover_vhid: null,
+      failover_group: 1,
+      failover_aliases: [
+        {
+          type: NetworkInterfaceAliasType.Inet,
+          address: '10.220.38.236',
+          netmask: 22,
+        },
+      ],
+      failover_virtual_aliases: [
+        {
+          type: NetworkInterfaceAliasType.Inet,
+          address: '10.220.39.70',
+          netmask: 32,
+        },
+      ],
+    },
+  ] as unknown as NetworkInterface[];
```

Comment:
Unknown is not needed. You could also remove properties that are not used by the test. Since you're only comparing objects to themselves, you can reduce them to `[{ id: 1 }, { id: 2 }] as NetworkInterface[]`.